diff --git a/hostapd/Makefile b/hostapd/Makefile
index 15d4070e1..7aac2c13d 100644
--- a/hostapd/Makefile
+++ b/hostapd/Makefile
@@ -192,6 +192,7 @@ CFLAGS += -DUCODE_SUPPORT
 OBJS += ../src/utils/ucode.o
 OBJS += ../src/ap/ucode.o
 NEED_ULOOP:=y
+LIBS += -lblobmsg_json -lucode -ludebug
 endif
 
 ifdef NEED_ULOOP
diff --git a/wpa_supplicant/Makefile b/wpa_supplicant/Makefile
index af0bceb2b..80c1afc35 100644
--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -197,6 +197,7 @@ OBJS += ../src/utils/ucode.o
 OBJS += ../src/utils/uloop.o
 OBJS += ucode.o
 LIBS += -lubox
+LIBS += -lblobmsg_json -lucode -ludebug
 endif
 
 ifdef CONFIG_CODE_COVERAGE
