From f02136395a9d534f3255a7ee1232aa9554c3e01c Mon Sep 17 00:00:00 2001
From: mtk27745 <rex.lu@mediatek.com>
Date: Tue, 14 Nov 2023 16:40:57 +0800
Subject: [PATCH] rdkb uc script support

---
 common.uc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common.uc b/common.uc
index 4c33779..ee30df9 100644
--- a/common.uc
+++ b/common.uc
@@ -49,6 +49,8 @@ function __phy_is_fullmac(phyidx)
 {
 	let data = nl80211.request(nl80211.const.NL80211_CMD_GET_WIPHY, 0, { wiphy: phyidx });
 
+	if (data == null)
+		return 1;
 	return !data.software_iftypes.ap_vlan;
 }
 
-- 
2.18.0

