From d8375d65c9a32bef94c44f8ec4fc689aaebc70d5 Mon Sep 17 00:00:00 2001
From: "howard.hsu" <howard-yh.hsu@mediatek.com>
Date: Fri, 8 Apr 2022 14:37:43 +0800
Subject: [PATCH] HAL: add wifi_getRadioStatsEnable()

---
 source/wifi/wifi_hal.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/source/wifi/wifi_hal.c b/source/wifi/wifi_hal.c
index 2e60367..a70ff15 100644
--- a/source/wifi/wifi_hal.c
+++ b/source/wifi/wifi_hal.c
@@ -9652,3 +9652,11 @@ INT wifi_getProxyArp(INT apIndex, BOOL *enable)
     wifi_dbg_printf("\n[%s]: proxy_arp is : %s", __func__, output);
     return RETURN_OK;
 }
+
+INT wifi_getRadioStatsEnable(INT radioIndex, BOOL *output_enable)
+{
+    if (NULL == output_enable || !(radioIndex==0 || radioIndex==1))
+        return RETURN_ERR;
+    *output_enable=TRUE;
+    return RETURN_OK;
+}
-- 
2.18.0

