From fe9452e2d7c27c7ae680b32a4db9f92f90d4ca4b Mon Sep 17 00:00:00 2001
From: Allen Ye <allen.ye@mediatek.com>
Date: Thu, 14 Apr 2022 19:10:15 +0800
Subject: [PATCH] HAL: refactor wifi_factoryResetRadio()

---
 source/wifi/wifi_hal.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/source/wifi/wifi_hal.c b/source/wifi/wifi_hal.c
index bfa7b01..8b302d5 100644
--- a/source/wifi/wifi_hal.c
+++ b/source/wifi/wifi_hal.c
@@ -726,15 +726,17 @@ INT wifi_factoryResetRadios()
 */
 INT wifi_factoryResetRadio(int radioIndex) 	//RDKB
 {
+    system("systemctl stop hostapd.service");
+
     WIFI_ENTRY_EXIT_DEBUG("Inside %s:%d\n",__func__, __LINE__);
     if(radioIndex == 0)
-        system("cp /etc/hostapd-2G.conf /nvram/hostapd0.conf");
+        system("rm /nvram/hostapd0.conf");
     else if(radioIndex == 1)
-        system("cp /etc/hostapd-5G.conf /nvram/hostapd1.conf");
+        system("rm /nvram/hostapd1.conf");
     else
          return RETURN_ERR;
 
-    system("systemctl restart hostapd.service");
+    system("systemctl start hostapd.service");
     WIFI_ENTRY_EXIT_DEBUG("Exiting %s:%d\n",__func__, __LINE__);
     return RETURN_OK;
 }
-- 
2.18.0

