From b66aba3c8fc50e70a67f63e52b54ae687b90b45c Mon Sep 17 00:00:00 2001
From: Rex Lu <rex.lu@mediatek.com>
Date: Mon, 10 Nov 2025 14:28:57 +0800
Subject: [PATCH] test

---
 flowtable.sh | 2 +-
 smp-mt76.sh  | 9 +++++----
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/flowtable.sh b/flowtable.sh
index 8ac1d7c..e82ffbb 100644
--- a/flowtable.sh
+++ b/flowtable.sh
@@ -38,7 +38,7 @@ nftables_flowoffload_enable()
 	fi
 
 	# check all of the ETH/WiFi virtual interfaces
-	eth_ifs=$(ls /sys/class/net | grep -E '^(eth|lan|wan|mxl|ppp)')
+	eth_ifs=$(ls /sys/class/net | grep -E '^(eth|lan|wan|mxl|ppp|erouter)')
 	wifi_ifs=$(iw dev | grep 'Interface' | awk '{print $2}')
 
 	interfaces=$(echo -e "$eth_ifs\n$wifi_ifs")
diff --git a/smp-mt76.sh b/smp-mt76.sh
index abc4944..ccf1d5b 100644
--- a/smp-mt76.sh
+++ b/smp-mt76.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-source /sbin/flowtable.sh
+source /usr/sbin/flowtable.sh
 
 KERNEL_VERSION=$(uname -r)
 
@@ -9,7 +9,8 @@ CPU_LIST=`cat /proc/interrupts | sed -n '1p'`
 NUM_OF_CPU=0; for i in $CPU_LIST; do NUM_OF_CPU=`expr $NUM_OF_CPU + 1`; done;
 DEFAULT_RPS=0
 
-. /lib/functions.sh
+#. /lib/functions.sh
+board_name=`cat /proc/device-tree/compatible`
 
 RPS_IF_LIST=""
 NUM_WIFI_CARD=0
@@ -29,7 +30,7 @@ get_if_info()
 	NET_IF_LIST=`ls /sys/class/net`
 	for vif in $NET_IF_LIST;
 	do
-		if [[ "$vif" == "eth"* ]] ||  \
+		if [[ "$vif" == "eth"* ]] || [[ "$vif" == "erouter"* ]] || \
 		[[ "$vif" == "lan"* ]] || [[ "$vif" == "wan"* ]] || \
 		[[ "$vif" == "wlan"* ]] || [[ "$vif" == "phy"* ]] || \
 		[[ "$vif" == *"mld"* ]]; then
@@ -355,7 +356,7 @@ MT7622()
 
 setup_model()
 {
-	board=$(board_name)
+	board=$board_name
 	num_of_wifi=$NUM_WIFI_CARD
 
 	if [[ $board == *"7988"* ]]; then
-- 
2.45.2

