diff --git a/flowtable.sh b/flowtable.sh
index add1407..bc2e100 100644
--- a/flowtable.sh
+++ b/flowtable.sh
@@ -42,7 +42,7 @@ nftables_flowoffload_enable()
 	fi
 
 	# check all of the ETH/WiFi virtual interfaces
-	eth_ifs=$(ls /sys/class/net | grep -E '^(eth|lan|wan|mxl|ppp|464)')
+	eth_ifs=$(ls /sys/class/net | grep -E '^(eth|lan|wan|mxl|ppp|464|erouter)')
 	wifi_ifs=$(iw dev | grep 'Interface' | awk '{print $2}')
 
 	interfaces="$eth_ifs $wifi_ifs"
diff --git a/smp-mt76.sh b/smp-mt76.sh
index cd1b0de..e6df113 100644
--- a/smp-mt76.sh
+++ b/smp-mt76.sh
@@ -4,14 +4,15 @@
 # Copyright (C) 2022 MediaTek Inc.
 #
 
-source /sbin/flowtable.sh
+source /usr/sbin/flowtable.sh
 
 OPTIMIZED_FOR="$1"
 CPU_LIST=`cat /proc/interrupts | sed -n '1p'`
 NUM_OF_CPU=0; for i in $CPU_LIST; do NUM_OF_CPU=`expr $NUM_OF_CPU + 1`; done;
 DEFAULT_RPS=0
 
-. /lib/functions.sh
+#. /lib/functions.sh
+board_name=`cat /proc/device-tree/compatible`
 
 RPS_IF_LIST=""
 NUM_WIFI_CARD=0
@@ -31,7 +32,7 @@ get_if_info()
 	NET_IF_LIST=`ls /sys/class/net`
 	for vif in $NET_IF_LIST;
 	do
-		if [[ "$vif" == "eth"* ]] ||  \
+		if [[ "$vif" == "eth"* ]] || [[ "$vif" == "erouter"* ]] || \
 		[[ "$vif" == "lan"* ]] || [[ "$vif" == "wan"* ]] || \
 		[[ "$vif" == "wlan"* ]] || [[ "$vif" == "phy"* ]] || \
 		[[ "$vif" == *"mld"* ]]; then
@@ -357,7 +358,7 @@ MT7622()
 
 setup_model()
 {
-	board=$(board_name)
+	board=$board_name
 	num_of_wifi=$NUM_WIFI_CARD
 
 	if [[ $board == *"7988"* ]]; then
