diff --git a/ated.sh b/ated.sh
index b4f3ca0..571d8af 100644
--- a/ated.sh
+++ b/ated.sh
@@ -1,4 +1,4 @@
-#!/bin/ash
+#!/bin/bash
 # SPDX-License-Identifier: GPL-2.0-only
 #
 # Copyright (C) 2022 MediaTek Inc.
diff --git a/iwpriv.sh b/iwpriv.sh
index ffa25c8..3cd6371 100644
--- a/iwpriv.sh
+++ b/iwpriv.sh
@@ -1,4 +1,4 @@
-#!/bin/ash
+#!/bin/bash
 # SPDX-License-Identifier: GPL-2.0-only
 #
 # Copyright (C) 2022 MediaTek Inc.
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6137dfd..806db20 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -17,6 +17,7 @@ INCLUDE_DIRECTORIES("${backports_dir}/mac80211/uapi")
 
 ADD_EXECUTABLE(atenl main.c eth.c hqa.c nl.c eeprom.c util.c)
 TARGET_LINK_LIBRARIES(atenl nl-tiny)
+TARGET_LINK_LIBRARIES(atenl blkid)
 
 SET(CMAKE_INSTALL_PREFIX /usr)
 
