diff -urN a/debugfs.c b/debugfs.c
--- a/debugfs.c	2022-12-15 16:44:17.882701775 +0800
+++ b/debugfs.c	2022-12-15 16:44:34.686406545 +0800
@@ -32,10 +32,10 @@
 
 	if (!mt76_is_mmio(dev))
 		return -EOPNOTSUPP;
-
+#if 0
 	if (dev->napi_dev.threaded != val)
 		return dev_set_threaded(&dev->napi_dev, val);
-
+#endif
 	return 0;
 }
 
diff -urN a/mt7996/mcu.c b/mt7996/mcu.c
--- a/mt7996/mcu.c	2022-12-15 16:42:14.368885449 +0800
+++ b/mt7996/mcu.c	2022-12-15 16:43:30.047544516 +0800
@@ -9,6 +9,7 @@
 #include "mcu.h"
 #include "mac.h"
 #include "eeprom.h"
+#include <linux/moduleparam.h>
 
 struct mt7996_patch_hdr {
 	char build_date[16];
