diff --git a/switch_fun.c b/switch_fun.c
index 9e9f4fe..2082ba1 100644
--- a/switch_fun.c
+++ b/switch_fun.c
@@ -1435,6 +1435,24 @@ static void table_dump_internal(int type)
 				printf("%03x:   ", (value >> 16) & 0xfff);
 				reg_read(REG_ATRD_ADDR, &value2);
 				j = (value2 >> 4) & 0xff;	//r_port_map
+
+				if(j & 0x01)
+					printf("0");
+				else if (j & 0x02)
+					printf("1");
+				else if (j & 0x04)
+					printf("2");
+				else if (j & 0x08)
+					printf("3");
+				else if (j & 0x10)
+					printf("4");
+				else if (j & 0x20)
+					printf("5");
+				else if (j & 0x40)
+					printf("6");
+				else if (j & 0x80)
+					printf("7");
+				/*
 				printf("%c", (j & 0x01) ? '1' : '-');
 				printf("%c", (j & 0x02) ? '1' : '-');
 				printf("%c", (j & 0x04) ? '1' : '-');
@@ -1443,7 +1461,8 @@ static void table_dump_internal(int type)
 				printf("%c", (j & 0x20) ? '1' : '-');
 				printf("%c", (j & 0x40) ? '1' : '-');
 				printf("%c", (j & 0x80) ? '1' : '-');
-
+				*/
+				printf("       ");
 				reg_read(REG_TSRA2_ADDR, &mac2);
 
 				printf("   %2d", (mac2 >> 12) & 0x7);	//FID
diff --git a/switch_ioctl.h b/switch_ioctl.h
index dffe9c7..72cbee6 100644
--- a/switch_ioctl.h
+++ b/switch_ioctl.h
@@ -5,8 +5,8 @@
 #ifndef SWITCH_IOCTL_H
 #define SWITCH_IOCTL_H
 
-#define ETH_DEVNAME "eth0"
-#define BR_DEVNAME "br-lan"
+#define ETH_DEVNAME "eth1"
+#define BR_DEVNAME "brlan0"
 
 #define RAETH_MII_READ                  0x89F3
 #define RAETH_MII_WRITE                 0x89F4
