From b7cd3dc333501f11b1e70bbdd5ea8b6929d38f40 Mon Sep 17 00:00:00 2001
From: "Allen.Ye" <allen.ye@mediatek.com>
Date: Fri, 13 Jan 2023 13:22:27 +0800
Subject: [PATCH] Add owe mode

---
 wifi_hal_ap.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/wifi_hal_ap.h b/wifi_hal_ap.h
index 9eebe10..15c24f9 100644
--- a/wifi_hal_ap.h
+++ b/wifi_hal_ap.h
@@ -2455,7 +2455,8 @@ typedef enum {
     wifi_security_mode_wpa_wpa2_enterprise = 0x00000100,
     wifi_security_mode_wpa3_personal = 0x00000200,
     wifi_security_mode_wpa3_transition = 0x00000400,
-    wifi_security_mode_wpa3_enterprise = 0x00000800
+    wifi_security_mode_wpa3_enterprise = 0x00000800,
+    wifi_security_mode_enhanced_open = 0x00001000
 } wifi_security_modes_t;
 
 typedef struct {
-- 
2.18.0

